### PR TITLE
MTM-53068 improvements and refresh in microservice sdk docs.

### PR DESCRIPTION
**Pull request by jkot-softwareag**
- security general aspects (service and bootstap users, owner and subscriber tenants)
- up to date default jvm arguments
- additional notes in general development area

### CODE REVIEW

File: content/microservice-sdk/rest-bundle/microservice-development.md [hunk 1]
```diff
@@ -624,3 +627,6 @@ HTTP/1.1 200 OK
     }
 ]
 ```
+
+Note: All devices channel (*) is deprecated and should be avoided. Subscription on this channel may cause performance issues
```
Comment (slc-softwareag) [inline content/microservice-sdk/rest-bundle/microservice-development.md:631]: :+1:

**Review by slc-softwareag — APPROVED** [review]

**Review by d-mariak — APPROVED** [review]

File: content/microservice-sdk/rest-bundle/microservice-development.md (hunk 1 above)
Comment (pawelkowalczyksag) [inline content/microservice-sdk/rest-bundle/microservice-development.md:631]:
```suggestion
Note: A channel for all devices (*) is deprecated and should be avoided. Subscription on this channel may cause performance issues 
```
I find it a bit weirdly worded. How about this?

**Review by pawelkowalczyksag — APPROVED** [review]

File: content/microservice-sdk/concept-bundle/security.md [hunk 2]
```diff
@@ -112,18 +112,18 @@ In general, microservices use the standard {{< product-c8y-iot >}} authenticatio
...
 2. For accessing customer tenants, the microservice installs a set of required permissions for being able to operate.
 
-A microservice is associated with a service user in the {{< management-tenant >}}, which will make sure that only its subscriptions are returned. A microservice is also associated with a set of permissions that it requires for carrying out its function on a customer tenant.
-These permissions are visualized in the Administration application. The permissions are associated with the service user that is created when a platform administration associates a microservice with a tenant.
+A microservice is associated with a bootstrap user in the owner tenant, which will make sure that only its subscriptions are returned. A microservice is also associated with a set of permissions that it requires for carrying out its function on a customer tenant.
+These permissions are visualized in the Administration application. The permissions are associated with the service user that is created when a platform administration associates a microservice with a tenant (subscribes it). Owner tenant also needs to be subscribed if you intend to use microservice features on this tenant.
```
Comment (rssag) [inline content/microservice-sdk/concept-bundle/security.md:126]:
```suggestion
These permissions are visualized in the Administration application. The permissions are associated with the service user that is created when a platform administration associates a microservice with a tenant (subscribes to it). The owner tenant must also be subscribed to if you intend to use microservice features on it. 
```

File: content/microservice-sdk/java-bundle/developing-microservice.md [hunk 3]
```diff
@@ -599,7 +599,7 @@ For external/legacy deployment, the following paths will be searched in order to
 #### Logging
 
 For external/legacy deployment, logging into the application implies using [Spring Logging](https://docs.spring.io/spring-boot/docs/current/reference/html/howto-logging.html).
-The following locations are searched for log-back file:
+The following locations are searched for Logback configuration file:
```
Comment (rssag) [inline content/microservice-sdk/java-bundle/developing-microservice.md:602]:
```suggestion
The following locations are searched for the Logback configuration file:
```

File: content/microservice-sdk/rest-bundle/microservice-development.md [hunk 4]
```diff
@@ -388,6 +388,9 @@ Content-Type: application/vnd.com.nsn.cumulocity.managedobjectreference+json
 }
 ```
 
+Note: there is a possibility to link multiple child assets or multiple child devices to one parent in single request. 
+Refer to the [Inventory Child Operations](https://{{< domain-c8y >}}/api/core/{{< c8y-current-version >}}/#tag/Child-operations) in the {{< openapi >}} for more details.
```
Comment (rssag) [inline content/microservice-sdk/rest-bundle/microservice-development.md:392]:
```suggestion
{{< c8y-admon-info >}}
You can link multiple child assets or multiple child devices to the same parent in a single request. 
Refer to inventory child operations in the [Inventory API](https://{{< domain-c8y >}}/api/core/{{< c8y-current-version >}}/#tag/Child-operations) in the {{< openapi >}} for more details.
{{< /c8y-admon-info >}}
```

**Review by rssag — CHANGES_REQUESTED** [review]

**Comment by rssag** [comment]
what are the target versions?

File: content/microservice-sdk/rest-bundle/microservice-development.md (hunk 1 above)
Comment (jkot-softwareag) [reply content/microservice-sdk/rest-bundle/microservice-development.md:631]: (*) means the channel ending with *, which means listening to notifications from all devices and not a single one.
Examples
/events/1331421   - channel for notifications of events from device 1331421
/events/*   - channel for notifications of events from all devices (may be too much for subscribed client to handle)

**Comment by jkot-softwareag** [comment]
> what are the target versions?

I have added target versions, from 10.16 up. In fact I think all of this is truth for longer, but 10.16 will be soon released as docs and version installed on cumulocity cloud environments.

File: content/microservice-sdk/concept-bundle/security.md [hunk 5]
```diff
@@ -112,18 +112,18 @@ In general, microservices use the standard {{< product-c8y-iot >}} authenticatio
 1. The microservice can be created in any tenant that has **feature-microservice-hosting** enabled.
 2. The microservices access the tenant API.
 
-At the installation time of the microservice, an application is created in the {{< management-tenant >}} reflecting the new microservice. In addition, a service user is created in the {{< management-tenant >}} that allows the microservice to retrieve subscriptions.
+At the installation time of the microservice, an application is created in the owner tenant (usually "super-tenant") reflecting the new microservice. In addition, a bootstrap user is created in the owner tenant that allows the microservice to retrieve subscriptions.
```
Comment (BeateRixen) [inline content/microservice-sdk/concept-bundle/security.md:115]: We do not use the term "super tenant" as this is unclear. Could we say "usually the {{< management-tenant >}}"?

File: content/microservice-sdk/java-bundle/developing-microservice.md [hunk 6]
```diff
@@ -599,7 +599,7 @@ For external/legacy deployment, the following paths will be searched in order to
 #### Logging
 
 For external/legacy deployment, logging into the application implies using [Spring Logging](https://docs.spring.io/spring-boot/docs/current/reference/html/howto-logging.html).
-The following locations are searched for log-back file:
+The following locations are searched for the Logback configuration file:
```
Comment (BeateRixen) [inline content/microservice-sdk/java-bundle/developing-microservice.md:602]:
```suggestion
The following locations are searched for the logback configuration file:
```

File: content/microservice-sdk/rest-bundle/microservice-development.md [hunk 7]
```diff
@@ -624,3 +629,8 @@ HTTP/1.1 200 OK
     }
 ]
 ```
+
+{{< c8y-admon-info >}}
+A channel for all devices (*) is deprecated and should be avoided. Subscription on this channel may cause performance issues
+on both the client and the server side, and this options may be removed in future versions of {{< product-c8y-iot >}}.
```
Comment (BeateRixen) [inline content/microservice-sdk/rest-bundle/microservice-development.md:635]:
```suggestion
on both the client and the server side, and this option may be removed in future versions of {{< product-c8y-iot >}}.
```

File: content/microservice-sdk/java-bundle/developing-microservice.md (hunk 6 above)
Comment (jkot-softwareag) [reply content/microservice-sdk/java-bundle/developing-microservice.md:602]: "logback" is the library, like "Spring". It is up to you to decide @BeateRixen. For Spring we usually use big letter, for Logback I am not sure.

File: content/microservice-sdk/concept-bundle/security.md (hunk 5 above)
Comment (jkot-softwareag) [reply content/microservice-sdk/concept-bundle/security.md:115]: well, somebody told me "super-tenant" is "management or EE tenant" when I wanted to remove it from another place in the code. The problem with previous explanation was a claim that only management can create microservices, and in fact every tenant with subscibed feature-microservice-hosting can do it. Many EE tenants use this feature and microservices are subscribed to their subtenants.

Maybe 
`(usually management or Enterprise Edition tenant)` 
?

Comment (BeateRixen) [reply content/microservice-sdk/concept-bundle/security.md:115]: I would then phrase it "usually a {{< management-tenant >}} or {{< enterprise-tenant >}}". We use short codes for the tenant types. Btw, it is just Enterprise tenant, not Enterprise edition tenant :-).

I will change the terminology in the other place as well then, thanks for the link. We should be really clear with these terms.

File: content/microservice-sdk/rest-bundle/microservice-development.md (hunk 7 above)
Comment (jkot-softwareag) [reply content/microservice-sdk/rest-bundle/microservice-development.md]: done and improvement squashed with previous commit

**Review by BeateRixen — APPROVED** [review]

**Pull request merged**